arch/x86/kvm/pkvm/mmu.c

```diff
@@ -180,6 +180,26 @@ static int fix_hyp_mmu_page_refcnt(void)
 	return pkvm_pgtable_walk(&hyp_mmu, 0, size, &walker);
 }
 
+static void set_host_mem_pgstate(unsigned long phys, unsigned long size,
+				 enum pkvm_page_state pgstate)
+{
+	unsigned long end = PAGE_ALIGN(phys + size);
+	struct pkvm_page *page;
+
+	/*
+	 * PKVM_PAGE_STATE_BITS represents the size of enum pkvm_page_state. And
+	 * pkvm_page->host_state is limited 8 bits. Ensure pkvm_page->host_state
+	 * is large enough to store enum pkvm_page_state. If it is not, trigger
+	 * a build bug.
+	 */
+	BUILD_BUG_ON(PKVM_PAGE_STATE_BITS > 8);
```

Comment:
Any reason why add this BUILD_BUG_ON here, not after the `struct pkvm_page` definition in memory.h?

Also how about making the comment shorter, e.g. just "Make sure pkvm_page->host_state is large enough to store enum pkvm_page_state."?

(In particular, "If it is not, trigger a build bug." seems completely redundant, right?)

Reply:
Yeah, for this case, using static_assert also works and can also put it just after pkvm_page definition. I can change to that.